src/ipc/handler.c

```diff
@@ -648,10 +649,15 @@ static int ipc_pm_core_enable(uint32_t header)
 
 	for (i = 0; i < PLATFORM_CORE_COUNT; i++) {
 		if (i != PLATFORM_MASTER_CORE_ID) {
-			if (pm_core_config.enable_mask & (1 << i))
+			if (pm_core_config.enable_mask & (1 << i)) {
+				pm_runtime_put(CORE_MEMORY_POW, i);
 				cpu_enable_core(i);
-			else
+			}
+
+			else {
+				pm_runtime_get(CORE_MEMORY_POW, i);
```

Comment:
I think firstly we should disable core and then turn off memory used by this core